### ISSUE TITLE
feat: use prepare script to build the package

### ISSUE THREAD

**Issue by EvilG-MC** [Tomato6966/lavalink-client#45]
I noticed that the `dist` directory has unused code that is just sitting there.

It can easily be deleted and re-created, but I was thinking of implementing the `prepare` script so that when the github version is used, it automatically builds the package in some update, since it always deletes the `dist` and re-creates it.

```jsonc
"scripts": {
    "build": "npm run build:cjs && npm run build:esm && npm run build:types",
    "build:cjs": "node tools/cleanup cjs && tsc -p config/tsconfig.cjs.json",
    "build:esm": "node tools/cleanup esm && tsc -p config/tsconfig.esm.json",
    "build:types": "node tools/cleanup types && tsc -p config/tsconfig.types.json",
    "clean": "node tools/cleanup",
    
    "prepare": "npm run build" //<--- It will be executed each time the package is installed.
  }
```
It will be much more comfortable at the time of installation since they will be clean installations.

**Comment by Tomato6966** [comment]
The package does only release the dist and not the drc build is done on my machine not on urs therefore it should be cleaned up by myself too but idk why it isnt apparently? Imma look into it next time

**Comment by EvilG-MC** [comment]
There are files that have been there for a year.

- [structures](https://github.com/Tomato6966/lavalink-client/tree/main/dist/structures)
- [index.js](https://github.com/Tomato6966/lavalink-client/blob/main/dist/index.js)
- [index.d.ts](https://github.com/Tomato6966/lavalink-client/blob/main/dist/index.d.ts)

It seemed like a good idea to me anyway, especially for the github versions, since the release in npmjs ignores that script.

**Comment by Tomato6966** [comment]
I fixed it, it was because the clean code only cleaned the folders and never cleaned the build which was before the folder specific build release oops
released v2.3.1 which includes your types pr
